sources/core/Stride.Core.Mathematics/Int3.cs

```diff
@@ -551,7 +541,32 @@ public static Int3 Min(Int3 left, Int3 right)
             Min(ref left, ref right, out result);
             return result;
         }
-   
+
+        /// <summary>
+        /// Returns a vector containing the rounded values of the specified vector.
+        /// </summary>
+        /// <param name="value">The source vector.</param>
+        /// <param name="result">When the method completes, contains an new vector composed of the rounded values of the specified vector.</param>
+        /// <param name="rounding">The rounding strategy to use.</param>
+        public static void Round(ref Vector3 value, out Int3 result, MidpointRounding rounding = default)
+        {
+            result.X = (int)Math.Round(value.X, rounding);
+            result.Y = (int)Math.Round(value.Y, rounding);
+            result.Z = (int)Math.Round(value.Z, rounding);
+        }
```

Comment:
Replace `ref Vector3 value` with `in Vector3 value`
`MathF` has an overload for midpoint rounding see https://learn.microsoft.com/en-us/dotnet/api/system.mathf.round?view=net-7.0

Reply:
> Replace ref Vector3 value with in Vector3 value

Those should be wider changes throughout the whole codebase. A lot of methods don't use the `in` keyword. For me that's outside this PR which is mostly a build fix.

> MathF has an overload for midpoint rounding

Should I use MathF for the other places in that file (e.g. Sqrt and Pow)?

Reply:
> For me that's outside this PR
Fixing it later would be a breaking change.

>  A lot of methods don't use the in keyword.
I took care of a bunch of them a while ago but not all of them, no. So there's a bit of a mix right now.

> Should I use MathF for the other places in that file (e.g. Sqrt and Pow)?
Yes please, as long as you touched those lines in this PR, we'll leave the rest of the file for another PR.

Reply:
Seems reasonable.

Reply:
For Round I will use `MathF`. However for `Length()` and `LengthUntruncated()` I think we should keep using `Math`. The rationale is that it is doing a square root of a multiplication of `int`. An `int` * `int` fits in a `long` and the equivalent floating point type is `double`. So to avoid overflows, we should only convert the result to `float` at the end.